scripts/kani-regression.sh

```diff
@@ -94,7 +94,24 @@ fi
 # Check codegen of firecracker
 time "$SCRIPT_DIR"/codegen-firecracker.sh
 
+# Test run 'cargo kani assess scan' on the existing workspace assess test
+echo "Running assess scan test:"
+(cd $KANI_DIR/tests/assess-scan-test-scaffold && \
+ cargo kani --enable-unstable assess scan && \
+ (cd foo && cargo clean) && \
+ (cd bar && cargo clean))
+EXPECTED_FILES=(bar/bar.kani-assess-metadata.json foo/foo.kani-assess-metadata.json bar/bar.kani-assess.log foo/foo.kani-assess.log)
+for file in ${EXPECTED_FILES[@]}; do
+  if [ -f $KANI_DIR/tests/assess-scan-test-scaffold/$file ]; then
```

Comment:
Should we check the contents of the file to make sure nothing is broken? Perhaps diff them against an expected file? Or grep for certain lines?

Reply:
I'm not happy with trying to add more ad-hoc testing steps here. There are at least a couple of checks implicitly happening here:

1. If scan exits unsuccessfully, it fails.
2. If something goes wrong with the individual assesses, the metadata files aren't emitted, even if scan somehow misses the failure.
3. The details of the results shouldn't actually be any different compared to the assess tests that are in the `cargo-kani` suite. It'd be nice to check that, but it'd kinda be crudely emulating an expect test but with greps?

I think we should do better, but we're stacking up ad-hoc testing scripts that really need a more principled solution...